bwc-test/src/test/java/SecurityBackwardsCompatibilityIT.java

```diff
@@ -61,6 +86,44 @@ protected final Settings restClientSettings() {
             .build();
     }
 
+    @Override
+    protected RestClient buildClient(Settings settings, HttpHost[] hosts) throws IOException {
+        RestClientBuilder builder = RestClient.builder(hosts);
+        configureHttpsClient(builder, settings);
+        boolean strictDeprecationMode = settings.getAsBoolean("strictDeprecationMode", true);
+        builder.setStrictDeprecationMode(strictDeprecationMode);
+        return builder.build();
+    }
+
+    protected static void configureHttpsClient(RestClientBuilder builder, Settings settings) throws IOException {
+        Map<String, String> headers = ThreadContext.buildDefaultHeaders(settings);
+        Header[] defaultHeaders = new Header[headers.size()];
+        int i = 0;
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
+            defaultHeaders[i++] = new BasicHeader(entry.getKey(), entry.getValue());
+        }
+        builder.setDefaultHeaders(defaultHeaders);
+        builder.setHttpClientConfigCallback(httpClientBuilder -> {
+            String userName = Optional.ofNullable(System.getProperty("tests.opensearch.username"))
+                .orElseThrow(() -> new RuntimeException("user name is missing"));
+            String password = Optional.ofNullable(System.getProperty("tests.opensearch.password"))
+                .orElseThrow(() -> new RuntimeException("password is missing"));
+            CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(userName, password));
+            try {
+                return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider)
+                    // disable the certificate since our testing cluster just uses the default security configuration
+                    .setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)
+                    .setSSLContext(SSLContextBuilder.create().loadTrustMaterial(null, (chains, authType) -> true).build());
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+
+        // final TimeValue socketTimeout = TimeValue.parseTimeValue("60s");
+        // builder.setRequestConfigCallback(conf -> conf.setSocketTimeout(Math.toIntExact(socketTimeout.getMillis())));
+    }
+
     public void testBasicBackwardsCompatibility() throws Exception {
```

Comment:
Building off of this, for the base case we should at least call the whoiam api to confirm that REST endpoint is registered and up.  Maybe call some APIs where a 403 is expected due to missing AuthZ.

Reply:
Should the additional tests be done in this PR? 

wdyt about writing a META issue about additional BWC tests to add?

Reply:
I am distrustful of these tests if we don't a validate any security scenarios plugin explicitly, these tests **passed** when the security plugin was disabled.  If we think that BWC testing scenarios are working, lets put them to the test at least a little.

Reply:
@peternied I added a very simple WhoAmI test. When the security plugin is disabled, it would give back a response that says the endpoint isn't registered and fail the test.

The test calls on the WhoAmI endpoint and looks for a key (`dn` in this case) in the response.

See below for call and response to whoami endpoint with admin basic internal user:

```
> curl -XGET https://admin:admin@localhost:9200/_plugins/_security/whoami --insecure
{"dn":null,"is_admin":false,"is_node_certificate_request":false}
```

FYI WhoAmI has logic to [skip authentication](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/filter/SecurityRestFilter.java#L248) which is why the response is sparse.

Reply:
haha, that output is beyond silly, but that is what our super admins look like